plugins/design.cpp

```diff
@@ -0,0 +1,271 @@
+#include <bitset>
+#include <cstdint>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "ColorText.h"
+#include "Debug.h"
+#include "LuaTools.h"
+#include "PluginManager.h"
+#include "df/graphic_viewportst.h"
+#include "df/world.h"
+#include "modules/Gui.h"
+#include "modules/Persistence.h"
+#include "modules/Screen.h"
+#include "modules/World.h"
+
+DFHACK_PLUGIN("design");
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+using DFHack::color_value;
+
+REQUIRE_GLOBAL(window_x);
+REQUIRE_GLOBAL(window_y);
+REQUIRE_GLOBAL(world);
+REQUIRE_GLOBAL(plotinfo);
+using namespace DFHack;
+using namespace df::enums;
+
+enum ConfigValues {
+  CONFIG_IS_ENABLED = 0,
+};
+
+namespace DFHack {
+    DBG_DECLARE(pathable, log, DebugCategory::LINFO);
+}
+
+DFhackCExport command_result plugin_init(color_ostream &out, std::vector<PluginCommand> &commands) {
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_shutdown(color_ostream &out) {
+    return CR_OK;
+}
+
+std::map<uint32_t, Screen::Pen> PENS;
```

Comment:
I believe it is possible for a world to have the cursors loaded as different texpos values (though I have never proved it). I suspect it would be a good idea to clear PENS on world unload:
```cpp
DFhackCExport command_result plugin_onstatechange(color_ostream &out, state_change_event event) {
    if (event == DFHack::SC_WORLD_UNLOADED) {
        DEBUG(log,out).print("clearing PENS\n");
        PENS.clear();
    }
    return CR_OK;
}
```